assets/js/live_vue/hooks.js

```diff
@@ -49,7 +49,7 @@ function getProps(el, liveSocket) {
     }
 }
 
-export function getHooks(components) {
+export function getHooks(components, optionalCallback) {
```

Comment:
It's great, but I'd do two changes:

1. Make second argument an object `options`, so in future we can put there more options instead of adding more arguments.
2. Rename `optionalCallback` to `initializeApp` and give the user power to customize the whole process? Maybe we could pass there not only app itself, but everything (component, element, props, slots, makeApp, plugin for injecting)? I kinda like the way how it looks in Inertia.js https://inertiajs.com/client-side-setup#initialize-the-inertia-app

So it could look like:

```javascript
// exported so it could be used for easy customization, like
// getHooks(components, { initializeApp: (context) => initializeVueApp(context).use(pinia) })
// but it would still allow for a totally custom initialization

export function initializeVueApp = ({createApp, component, props, slots, plugin, el}) => {
  return createApp({ render: () => h(component, props, slots) })
    .use(plugin)
    .mount(el)
}

// ...
export function getHooks(components, options = {}) {
  async mount(el, liveSocket) {
    ...
    const initializeApp = options.initializeApp || initializeVueApp
    el._instance = initializeApp({
      createApp: makeApp,
      component: component,
      props: this._props,
      slots: this._slots,
      plugin: { install: (app) => app.provide(liveInjectKey, this) },
      el: el
  }  
}
```

This is just a quick pseudo-code, but hopefully the idea is clear?